Optimize Database Queries : You should use a LIMIT clause on SQL queries. Use less data and limit it to the bare minimum.
 For example, the LIMIT clause limits the number of result rows in relational databases. When possible, using the 'LIMIT'
 clause reduces the amount of transferred data. Performance gains will be even more significant if records contain
 a large number of voluminous fields.

## Noncompliant Code Example

```java
 String sql = "SELECT user FROM myTable";
```

## Compliant Code Example

```java
 String sql = "SELECT user FROM myTable LIMIT 50";
```