Avoid declare variables in loops

## Noncompliant Code Example

```java
for(int i = 0; i < 5; i++){
    int x = i * 2;
}
```

```java
 int j = 0;
 while (j < 5) {
    int y = j * 3;
    j++;
 }
```

## Compliant Code Example

```java
int x;
for(int i = 0; i < 5; i++){
    x = i * 2;
}
```

```java

int y;
while (j < 5) {
    y = j * 3; // Compliant
    j++;
}
```